### ISSUE TITLE
Support injection of HttpClient instances

### ISSUE THREAD

**Issue by martincostello** [octokit/octokit.graphql.net#125]
The `Connection` class should be extensible enough to allow for interoperability with the [HttpClientFactory](https://github.com/aspnet/HttpClientFactory) in .NET Core 2.1.

This would allow users to use the out-of-the-box GraphQL client in their applications easily while applying cross-cutting concerns like retries, handler pooling, centralised configurability etc. without having to implement `IConnection` themselves to remove the `HttpClient` instance [created privately](https://github.com/octokit/octokit.graphql.net/blob/f100e9af01112a0e3226893275f31695750262e2/Octokit.GraphQL.Core/Connection.cs#L40) per `Connection` instance.

**Comment by StanleyGoldman** [comment]
That is a very good suggestion.

**Comment by grokys** [comment]
Would an additional ctor implemented like this be sufficient?

```
public Connection(HttpClient client, Uri uri, ICredentialStore credentialStore)
{
    Uri = uri;
    CredentialStore = credentialStore;
    HttpClient = client;
}
```

**Comment by martincostello** [comment]
Maybe also one where the `Uri` isn’t required, given that I guess that exists just for Enterprise? The code would also need to add the UserAgent if not already present, or would you just leave that completely up to the integrator?

Happy to submit a PR if there’s a rough approach agreed.

**Comment by StanleyGoldman** [comment]
We are pretty interactive...
If you start me off with a unit test that shows us what you are thinking...
I can either take over or point you in the right directions to finish.

**Comment by martincostello** [comment]
Cool, I'll give that a whirl when I get next a spare hour or so.

**Comment by martincostello** [comment]
Have opened #131 with some very basic changes to `Connection` with a test showing the sort of things that can be achieved by supporting `HttpClient` injection.

**Comment by martincostello** [comment]
I guess this is closed now #131 is merged? 😺

**Comment by StanleyGoldman** [comment]
Yup.
